### DISCUSSION TITLE
Compatibility with 4/8MHz Crystals

### DISCUSSION THREAD

**Discussion by ajesquirell** [Arduino-IRremote/Arduino-IRremote#790] [Q&A] [answered]
Hello! I couldn't find much information about this elsewhere, so hopefully someone here can confirm. Based on the IRremoteBoardDefs.h file, this library should work with a 4MHz or an 8MHz crystal oscillator with, for example, the ATMega328P - correct? This is a configuration I am considering, compared to using a 16MHz crystal (For power savings). It would be for Sending IR signals only.

If it works, by next concern is the quality of IR output using a 4/8MHz clock instead of 16MHz. Does anyone know if this would be a reliable configuration to use?

Something that makes me question this is based on the wiki for the KontroLIR remote which was recently posted about in this repo as a show and tell discussion. I have attached the screenshot, but it basically says 4MHz is desired for battery operation, but 16MHz is optimal for this IRremote library...

Thanks!

![Capture](https://user-images.githubusercontent.com/47311328/107477912-9ebcfb00-6b46-11eb-89d0-e8e40e4d0c20.PNG)

**Comment by AnalysIR** [comment, marked as answer]
If you have the choice , 16Mhz provides better timing accuracy(granularity) and thus better performance (using Arduino & IRremote) - as a general rule.

When running on battery, the voltage is usually lower than the rating for 16Mhz on AVRs so this should be avoided. (2xAAA is up to 3.2volts  which is below the rating for 16Mhz)

Currently, we are only supporting 4Mhz on IRremote up to 2.6.1. There have been a lot of changes in later versions & we will wait until things settle down more before considering adding in support for the latest v3+.

IMHO 4Mhz is the optimal for running on 2xAAA batteries.

If you have very good low power management configured in your firmware then 8MHz is also feasible as the battery will also last a long time (but less than on 4MHz which can run at a lower voltage level than 8MHz). You could expect 1->2 years+ for typical domestic usage.

FYI: If you use the lowest power mode for sleeping(PWRDWN mode), then all 3 clock speeds will draw the same 'low' current when sleeping. 

**Summary**: for 2xAA or 2xAAA 8MHz & 4MHz are good. 16MHz may work, but AVRs are not officially rated for this.

**Reply by ajesquirell** [reply]
Thanks so much!

**Comment by AnalysIR** [comment]
...forgot to mention, you can go with any crystal (16,8 or 4) and use clock division in either the fuses or firmware to get a lower clock speed).
